### PR TITLE
Add `GIFT_CARD_SENT` async event

### LINKED ISSUE #12165: Async event substitute for send_gift_card plugin event
### What I'm trying to achieve

Migrate customer emails plugin to the App. Currently, there's no available async event to `send_gift_card`. 

### Describe a proposed solution

Invoices already have a familiar behaving event - `INVOICE_SENT`.

### Other solutions I've tried and won't work

Gift card created doesn't seem to be the same event - creation and sending to the user are different actions.

### PR DESCRIPTION
**Pull request by rafalp**
Adds `GIFT_CARD_SENT` async event that's triggered when gift card is sent to somebody.

Fixes #12165

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/management/commands/get_graphql_schema.py [hunk 1]
```diff
@@ -8,4 +8,7 @@ class Command(BaseCommand):
     help = "Writes SDL for GraphQL API schema to stdout"
 
     def handle(self, *args, **options):
-        self.stdout.write(print_schema(schema))
+        # self.stdout.write(print_schema(schema))
+        with open("new-schema.graphql", "w") as fp:
```
Comment (fowczarek) [inline saleor/graphql/management/commands/get_graphql_schema.py:12]:
```suggestion
        with open("new-schema.graphql", "w") as fp:
```

**Review by fowczarek — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by jakubkuc — APPROVED** [review]

**Comment by maarcingebala** [comment]
@rafalp Can you please also update [the docs](https://docs.saleor.io/docs/3.x/developer/extending/apps/asynchronous-webhooks) with the new event? https://github.com/saleor/saleor-docs/

**Comment by rafalp** [comment]
@maarcingebala rebased. I can also update the docs. Is the event reference generated via some tool or maintained by hand?

**Comment by maarcingebala** [comment]
@rafalp By hand

**Comment by rafalp** [comment]
Docs PR: https://github.com/saleor/saleor-docs/pull/728

**Pull request merged**